src/SpectralLinter.php

```diff
@@ -58,7 +75,14 @@ public function getNpmPackageName() {
   }
 
   protected function getMandatoryFlags() {
-    return array('lint', '--format', 'json', '--quiet');
+    $flags = array('lint', '--format', 'json', '--quiet');
+
+    if ($this->ruleset) {
+      $flags[] = '--ruleset';
+      $flags[] = $this->ruleset;
+    }
```

Comment:
This should go in `getDefaultFlags`

Reply:
I think we should also keep `--ruleset` to one "flags" entry, as one never makes sense without the other.

Reply:
- `getDefaultFlags` is overwritten when `flags` is given in configuration, so it's just that: default flags. When `spectral.ruleset` is given, we _always_ want it to be applied, even when `flags` is also given, so this is the best place to do that.
- I thought the flags array is meant to represent space-separated command line arguments, for the purpose of per-option quoting. If it were `--result=file.yaml`, then it would be one "flag", but with the space, it's two "flags".

Reply:
All of that makes sense. It means we should modify some existing linters then because they are not following those rules.